sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -500,6 +501,32 @@ def _add_argparse_args(cls, parser):
                         choices=['COST_OPTIMIZED', 'SPEED_OPTIMIZED'],
                         help='Set the Flexible Resource Scheduling mode')
 
+  def _get_default_gcp_region():
+    """Get a default value for Google Cloud region according to
+    https://cloud.google.com/compute/docs/gcloud-compute/#default-properties.
+    If no other default can be found, returns 'us-central1'.
+    """
+    environment_region = os.environ.get('CLOUDSDK_COMPUTE_REGION')
+    if environment_region:
+      logging.info('Using default GCP region %s from $CLOUDSDK_COMPUTE_REGION',
+          environment_region)
+      return environment_region
+    try:
+      cmd = ['gcloud', 'config', 'get-value', 'compute/region']
+      output = subprocess.check_output(cmd).decode('utf-8').strip()
+      if '(unset)' not in output:
+        logging.info('Using default GCP region %s from `%s`',
+            output, ' '.join(cmd))
+        return output
+    except subprocess.CalledProcessError:
+      pass
+    logging.warning(
+        '--region not set; will default to us-central1. Future releases of '
+        'Beam will require the user to set --region explicitly, or else have a '
+        'default set via the gcloud tool. '
+        'https://cloud.google.com/compute/docs/regions-zones')
+    return 'us-central1'
```

Comment:
We should not default us-central1. If we cannot get this information, the flag should be required.

Reply:
Actually it makes sense with the warning above. Yes, we can default for now and will change in the future.